### PR TITLE
[Vmware] Add missing condition to cleanup nics if there are commands to send

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR allows cleaning up nics in case there are nicExpunge commands to be sent

Fixes: #5308  

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test matrix

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -636,7 +636,7 @@ protected void advanceExpunge(VMInstanceVO vm) throws ResourceUnavailableExcepti
 
         // send hypervisor-dependent commands before removing
         final List<Command> finalizeExpungeCommands = hvGuru.finalizeExpunge(vm);
-        if (finalizeExpungeCommands != null && finalizeExpungeCommands.size() > 0) {
+        if (CollectionUtils.isNotEmpty(finalizeExpungeCommands) || CollectionUtils.isNotEmpty(nicExpungeCommands)) {
             if (hostId != null) {
                 final Commands cmds = new Commands(Command.OnError.Stop);
                 for (final Command command : finalizeExpungeCommands) {
```
Comment (shwstppr) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:642]: @nvazquez can you please check if there is a need to check `finalizeExpungeCommands != null` here. With change, this code can be reached when `finalizeExpungeCommands` is null and `nicExpungeCommands` is having items and for KVM `finalizeExpunge` might return null

Comment (nvazquez) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java]: Thanks @shwstppr, refactored

**Comment by weizhouapache** [comment]
code lgtm

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 2]
```diff
@@ -667,6 +659,15 @@ protected void advanceExpunge(VMInstanceVO vm) throws ResourceUnavailableExcepti
 
     }
 
+    private void addAllExpungeCommandsFromList(List<Command> cmdList, Commands cmds, VMInstanceVO vm) {
+        if (CollectionUtils.isNotEmpty(cmdList)) {
```
Comment (GutoVeronezi) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:663]: We could invert the logic here, to reduce code indentation, and maybe add a log in trace.

**Review by rhtyd — APPROVED** [review]
Lgtm, needs testing

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java (hunk 2 above)
Comment (nvazquez) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java]: Thanks @GutoVeronezi, done

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test matrix

**Review by GutoVeronezi — APPROVED** [review]
The code changes LGTM, though I did not test it.

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test matrix

**Pull request merged**